### PR TITLE
refactor CurtinInstallStep a bit

### PR DESCRIPTION
**Pull request by mwhudson**
Prep work for fde installs being ... different.

Not sure if this is the right final form but it's probably a step in the right direction.

### CODE REVIEW

File: subiquity/server/controllers/install.py [hunk 1]
```diff
@@ -210,45 +218,42 @@ async def curtin_install(self, *, context, source):
...
+                source=source,
+                acquire_config=acquire_config,
+            )
+
+        await make_curtin_step(
+            "initial",
+            stages=[],
+            acquire_config=self.acquire_initial_config).run(context=context)
 
         generic_steps = [
```
Comment (ogayot) [inline subiquity/server/controllers/install.py:239]: if we're storing callables here, I think we don't need to treat the "initial" step differently - and we can drop the _generic_ prefix from this variable.

**Review by ogayot — APPROVED** [review]
LGTM, just one suggestion to refactor further.

File: subiquity/server/controllers/install.py [hunk 2]
```diff
@@ -73,16 +73,49 @@ def feed(self, line):
             self.traceback.append(line)
 
 
+class InstallStep(Protocol):
```
Comment (ogayot) [inline subiquity/server/controllers/install.py:76]: We're not using this Protocol, are we?

Comment (mwhudson) [reply subiquity/server/controllers/install.py:76]: No, I did at one point but no longer.

File: subiquity/server/controllers/install.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/server/controllers/install.py:239]: Well one of the very next things to do is to start passing different configs to different invocations. Possibly a caller here isn't the best way to do it but it's sufficient I guess?

Comment (ogayot) [reply subiquity/server/controllers/install.py:232]: Yes, a caller/callable is fine IMO.

What I meant is that we could add have all the steps (including the initial step in the same sequence):

```diff
-         await make_curtin_step(
-           "initial",
-            stages=[],
-            acquire_config=self.acquire_initial_config).run(context=context)
-
-        generic_steps = [
+       steps: List[Callable[Context], None] = [
+              make_curtin_step(name="initial", stages=[], acquire_config=self.acquire_initial_config).run
               make_curtin_step(name="partitioning", stages=["partitioning"], acquire_config=self.acquire_generic_config).run
               make_curtin_step(name="extract", stages=["extract"], acquire_config=self.acquire_generic_config).run
```

Comment (mwhudson) [reply subiquity/server/controllers/install.py:232]: oh right, yes that makes sense

Comment (ogayot) [reply subiquity/server/controllers/install.py:227]: thanks!

**Pull request merged**